core/sql/exp/exp_conv.cpp

```diff
@@ -9321,6 +9321,32 @@ convDoIt(char * source,
   };
   break;
 
+// gb2312 -> utf8
+  case CONV_GBK_F_UTF8_V:
+  {
+    char * targetbuf = new char[sourceLen*4+1];
+    size_t sl = sourceLen;
+    int convLen = gbkToUtf8( source, sl, targetbuf, sl*4);
+    int copyLen = 0;
+    if (convLen > 0) {
+      copyLen = (convLen< targetLen) ? convLen: targetLen;
```

Comment:
My guess is that the targetbuf is always long enough, since it allows 4 bytes for every byte in the source. But you are right, there could be truncation when copyLen is calculated. I'm wondering if type synthesis for TRUNCATE guarantees that the output length is always large enough, though, and it is only upon assignment to a target column that we'd truncate?